### PR TITLE
Bundle dynamic extension frameworks

### PR DESCRIPTION
**Pull request by britzl**
When Cocoapod dependencies include dynamic frameworks these need to be signed and included in the application bundle. This change will copy any frameworks from the extender build result into the application bundle. Part of https://github.com/defold/extender/issues/274.

## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

### CODE REVIEW

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Bob.java [hunk 1]
```diff
@@ -383,22 +383,6 @@ public static List<File> getDefaultDmengineFiles(Platform platform, String varia
         return binaryFiles;
     }
 
-    public static List<File> getNativeExtensionEngineBinaries(Platform platform, String extenderExeDir) throws IOException
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Bob.java:386]: Moved to ExtenderUtil

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java [hunk 2]
```diff
@@ -180,19 +180,19 @@ public String getBuildDirectory() {
     }
 
     public String getPluginsDirectory() {
-        return FilenameUtils.concat(getRootDirectory(), PLUGINS_DIR);
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java:183]: Let's be consistent. We use rootDirectory everywhere else in Project.java

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/IOSBundler.java [hunk 3]
```diff
@@ -252,19 +252,15 @@ public void bundleApplication(Project project, Platform platform, File bundleDir
...
-
         // Loop over all architectures needed for bundling
         // Pickup each binary, either vanilla or from a extender build.
         List<File> binaries = new ArrayList<File>();
         for (Platform architecture : architectures) {
-            List<File> bins = Bob.getNativeExtensionEngineBinaries(architecture, extenderExeDir);
+            List<File> bins = ExtenderUtil.getNativeExtensionEngineBinaries(project, architecture);
             if (bins == null) {
                 bins = Bob.getDefaultDmengineFiles(architecture, variant);
-            } else {
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/IOSBundler.java:267]: This file will have a bunch of code formatting changes such as this one

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/IOSBundler.java [hunk 4]
```diff
@@ -437,21 +436,32 @@ public void bundleApplication(Project project, Platform platform, File bundleDir
         destExecutable.setExecutable(true);
         logger.info("Bundle binary: " + getFileDescription(destExecutable));
 
+        // Copy extension frameworks
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/IOSBundler.java:439]: The actual purpose of the PR

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/IOSBundler.java [hunk 5]
```diff
@@ -252,19 +252,15 @@ public void bundleApplication(Project project, Platform platform, File bundleDir
         final String variant = project.option("variant", Bob.VARIANT_RELEASE);
         final boolean strip_executable = project.hasOption("strip-executable");
 
-        // If a custom engine was built we need to copy it
-        boolean hasExtensions = ExtenderUtil.hasNativeExtensions(project);
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/IOSBundler.java:256]: Unused

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/IOSBundler.java [hunk 6]
```diff
@@ -252,19 +252,15 @@ public void bundleApplication(Project project, Platform platform, File bundleDir
...
-        File extenderPlatformDir = new File(project.getBuildDirectory(), architectures.get(0).getExtenderPair());
-        String extenderExeDir = FilenameUtils.concat(project.getRootDirectory(), "build");
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/IOSBundler.java:258]: This is equivalent to project.getBinaryOutputFolder()

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/IOSBundler.java [hunk 7]
```diff
@@ -437,21 +436,32 @@ public void bundleApplication(Project project, Platform platform, File bundleDir
         destExecutable.setExecutable(true);
         logger.info("Bundle binary: " + getFileDescription(destExecutable));
 
+        // Copy extension frameworks
+        logger.info("Copy frameworks");
```
Comment (JCash) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/IOSBundler.java:440]: logger.fine ?
No need to see this for every user?

**Comment by britzl** [comment]
Related PR: https://github.com/defold/defold/pull/7634

**Review by JCash — DISMISSED** [review]

**Pull request merged**